fvm/state/address_generator.go

```diff
@@ -12,42 +12,66 @@ const keyAddressState = "account_address_state"
 // It uses the underlying generator it gets from the chain.
 // The only change is that when next address is called the state is updated as well.
 type StateBoundAddressGenerator struct {
-	generator flow.AddressGenerator
-	state     *State
+	stateHolder *StateHolder
+	chain       flow.Chain
 }
 
-func NewStateBoundAddressGenerator(state *State, chain flow.Chain) (*StateBoundAddressGenerator, error) {
-	stateBytes, err := state.Get("", "", keyAddressState)
+func NewStateBoundAddressGenerator(stateHolder *StateHolder, chain flow.Chain) (*StateBoundAddressGenerator, error) {
+	return &StateBoundAddressGenerator{
+		stateHolder: stateHolder,
+		chain:       chain,
+	}, nil
```

Comment:
We are always returning a `nil` for this factory. Removing it saves some error checking efforts at the caller side.